src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -585,18 +585,17 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
     public List<StudentAttributes> createStudents(String lines, String courseId) throws EnrollException {
         List<String> invalidityInfo = new ArrayList<>();
         String[] linesArray = lines.split(Const.EOL);
-        ArrayList<String> studentEmailList = new ArrayList<>();
         List<StudentAttributes> studentList = new ArrayList<>();
 
         StudentAttributesFactory saf = new StudentAttributesFactory(linesArray[0]);
 
         for (int i = 1; i < linesArray.length; i++) {
             String line = linesArray[i];
             String sanitizedLine = SanitizationHelper.sanitizeForHtml(line);
+            if (StringHelper.isWhiteSpace(line)) {
+                continue;
+            }
             try {
-                if (StringHelper.isWhiteSpace(line)) {
-                    continue;
-                }
                 StudentAttributes student = saf.makeStudent(line, courseId);
 
                 if (!student.isValid()) {
```

Comment:
Let's SLAP this i.e. extract this into a helper method `addInvalidStudentInfo(List<String> invalidityInfo, StudentAttributes student, String sanitizedLine)`.